joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/dependency/GradleDependencies.scala

```diff
@@ -102,114 +102,113 @@ object GradleDependencies {
+      if (hasAndroidPrefixGradleProperty) {
+        val validProjectNames = List(project.getName) ++ project.getChildren.getAll.asScala.map(_.getName)
+        logger.debug(s"Found Gradle projects: ${validProjectNames.mkString(",")}")
+        if (!validProjectNames.contains(projectName)) {
+          val validProjectNamesStr = validProjectNames.mkString(",")
+          Failure(
+            new Exception(
+              s"The provided Gradle project name `$projectName` is is not part of the valid project names: `$validProjectNamesStr`"
+            )
+          )
+        } else {
+          Success(info)
+        }
+      } else {
+        Success(info)
+      }
+    } catch {
+      case t: Throwable =>
+        Failure(new Exception(s"Caught exception while trying fetch Gradle project information: `$t`.", t))
+    }
+  }
+
   // fetch the gradle project information first, then invoke a newly-defined gradle task to copy the necessary jars into
   // a destination directory.
   private[dependency] def get(
     projectDir: Path,
     projectName: String,
     configurationName: String
-  ): collection.Seq[String] = {
-    val gradleProjectInfoOption =
-      try {
-        logger.info(s"Fetching Gradle project information for project at path `$projectDir`.")
-        val connection = GradleConnector
-          .newConnector()
-          .forProjectDirectory(projectDir.toFile)
-          .connect()
-        val buildEnv = connection.getModel[BuildEnvironment](classOf[BuildEnvironment])
-        val project  = connection.getModel[GradleProject](classOf[GradleProject])
-
-        val hasAndroidPrefixGradleProperty =
-          try {
-            val out                      = new ByteArrayOutputStream()
-            val gradlePropertiesTaskName = "properties"
-            connection
-              .newBuild()
-              .forTasks(gradlePropertiesTaskName)
-              .setStandardOutput(out)
-              .run()
-            out.close()
-            val gradleAndroidPropertyPrefix = "android."
-            !out.toString.split('\n').filter(_.startsWith(gradleAndroidPropertyPrefix)).isEmpty
-          } catch {
-            case t: Throwable =>
-              logger.warn("Caught exception while executing Gradle task named `properties`.", t)
-              false
-          }
-        val info = GradleProjectInfo(
-          buildEnv.getGradle.getGradleVersion,
-          project.getTasks.asScala.map(_.getName).toSeq,
-          hasAndroidPrefixGradleProperty
+  ): Try[collection.Seq[String]] = {
+    getGradleProjectInfo(projectDir, projectName) match {
+      case Success(projectInfo) if projectInfo.gradleVersionMajorMinor()._1 < 5 =>
+        Failure(new Exception("Unsupported Gradle version `" + projectInfo.gradleVersion + "`"))
+      case Success(projectInfo) =>
+        logger.info(s"Creating gradle init script...")
+        val destinationDir = Files.createTempDirectory(tempDirPrefix)
+        destinationDir.toFile.deleteOnExit()
```

Comment:
There's a few different ways to create temp files here... I'd suggest using the same everywhere, and would opt for https://github.com/pathikrit/better-files#temporary-files, i.e. 
```
File.usingTemporaryFile { tempFile =>
  //do something
}  // tempFile is auto deleted at the end of this block - even if an exception happens
```

Reply:
:+1: